src/Streamly/Internal/Data/Parser/ParserD/Types.hs

```diff
@@ -145,66 +145,69 @@ import Streamly.Internal.Data.Strict (Tuple3'(..))
+-- Whenever a 'Partial n' result is returned we first backtrack by @n@ elements
+-- in the input and then release any remaining backtracking buffer. Similarly,
+-- 'Continue n' backtracks to @n@ elements before the current position and
+-- starts feeding the input from that point for future invocations of the
+-- parser.
+--
+-- If parser is not yet done, we can use the @extract@ operation on the @state@
+-- of the parser to extract a result. If the parser has not yet yielded a
+-- result, the operation fails with a 'ParseError' exception. If the parser
+-- yielded a 'Partial' result in the past the last partial result is returned.
+-- Therefore, if a parser yields a partial result once it cannot fail later on.
+--
+-- The parser can never backtrack beyond the position where the last partial
+-- result left it at. The parser must ensure that the backtrack position is
+-- always after that.
 --
 -- /Internal/
 --
 {-# ANN type Step Fuse #-}
 data Step s b =
-      Yield Int s
-      -- ^ Trim the backtracking buffer keeping only most recent @n@ items.
-      -- @Yield n state@ indicates that the parser has yielded a new result
-      -- which is a point of no return. The result can be extracted using
-      -- @extract@. The driver drops the buffer except @n@ most recent
-      -- elements.  The rule is that if a parser has yielded at least once it
-      -- cannot return a failure result.
-
-      | YieldB Int s
-      -- ^ Go back by @n@ items in the input buffer and drop the rest.
-      -- @YieldB n state@ indicates that the parser has yielded a new result
-      -- which is a point of no return. The result can be extracted using
-      -- @extract@. The driver moves back the current position of the cursor by
-      -- @n@ elements and drops any buffer before that.
-
-     -- only rewind
-    | Skip Int s
-    -- ^ Go back by @n@ items in the input buffer.
-    -- @Skip n state@ indicates that the parser has consumed the current input
-    -- but no new result has been generated. A new @state@ is generated.
-    -- However, if we use @extract@ on @state@ it will generate a result from
-    -- the previous @Yield@.  When @n@ is non-zero it is a backward offset from
-    -- the current position in the stream from which the driver will replay the
-    -- input to the parser. The offset cannot be beyond the latest point of no
-    -- return created by @Yield@.
-
-    | Stop Int b
-    -- ^ Trim the backtracking buffer keeping only most recent @n@ items.
-    -- @Stop n result@ asks the driver to stop driving the parser because it
-    -- has reached a fixed point and further input will not change the result.
-    -- @n@ is the count of unused input elements which includes the element on
-    -- which 'Stop' occurred.
+        Partial Int s
+    -- ^ Partial result with an optional backtrack.
+    --
+    -- @PartialWith count state@ means a partial result is available which
```

Comment:
`PartialWith`?

Reply:
typo. I started with using PartialWith etc. as written in the issue. But then I thought I might be making things more confusing by having more constructors. So kept it like this for now.